packages/EchoFramework/src/hooks/useHas3dAccess.ts

```
@@ -0,0 +1,35 @@
+import { BaseError, useInstCode } from '@equinor/echo-core';
+import { useEffect, useState } from 'react';
+import { check3dPermissionForPlantCode, check3dPermissionOnInstallation } from '../utils/plantInfo';
+/**
+ * Hook for checking if user has access to models for a given asset
+ *
+ * @export
+ * @param {string} [plantCode] asset to check access for, if not supplied instCode from user settings
+ * @param {((error: Error | BaseError | unknown) => void)} [handleError] method that will handle if an error occurred
+ * @return {*}  {boolean} returns wether or not the user have access
+ */
+export default function useHas3dAccess(
+    plantCode?: string,
+    handleError?: (error: Error | BaseError | unknown) => void
+): boolean {
+    const [has3dAccess, setHas3dAccess] = useState(true);
+    const instCode = useInstCode();
+
+    useEffect(() => {
+        async function checkAccess(): Promise<void> {
+            try {
+                const result = plantCode
+                    ? await check3dPermissionForPlantCode(plantCode)
+                    : await check3dPermissionOnInstallation(instCode);
+                setHas3dAccess(result);
+            } catch (error) {
+                if (handleError) handleError(error);
+                setHas3dAccess(false);
+            }
+        }
+        if (instCode) checkAccess();
+    }, [instCode, plantCode, handleError]);
```

We shouldn't return handleError function which is an in-argument.

We could return the error itself, that way the outside can deal with the error however they want. It's a 3rd solution, but the drawback is that the outside can easily just ignore the error, and it's never reported. This is probably the easiest solution.

Let's discuss this a bit in the office.

---

my mistake about handleError here, thought this array was the returned values for the hook. But I see now this is the boolean just below :)